presto-native-execution/presto_cpp/main/tests/DynamicFunctionRegistererTest.cpp

```diff
@@ -0,0 +1,209 @@
+    std::ofstream{dummySharedLibPathStr}.put('a');
+
+    // Create dummy shared library to emulate the shared library path
+    // validation.
+    auto filePathForDummySharedFile = exec::test::TempFilePath::create();
+    std::filesystem::path sharedFilePath(filePathForDummySharedFile->getPath());
+    auto dummySharedFile = sharedFilePath.replace_extension(kFileExtension);
+
+    writeToFile(pathStr, json);
+  }
+
+  void assertFunctionExistsAfter(
+      const facebook::velox::FunctionSignatureMap& before,
+      const facebook::velox::FunctionSignatureMap& after,
+      const std::string& name) {
+    EXPECT_TRUE(before.find(name) == before.end());
+    EXPECT_TRUE(after.find(name) != after.end());
+  }
+
+ private:
+  std::shared_ptr<SystemConfig> systemConfig;
+};
+
+TEST_F(DynamicFunctionRegistererTest, twoFunctions) {
+  std::string_view json = R"(
+  {
+    "dynamicUdfSignatureMap": {
+      "tmp": {
+        "mock1": [
+          {
+            "entrypoint": "registry123",
+            "fileName" : "test",
+            "outputType": "boolean",
+            "paramTypes": [
+            ]
+          }
+        ],
+        "mock2": [
+          {
+            "fileName" : "test",
+            "outputType": "varchar",
+            "paramTypes": [
+              "varchar"
+            ]
+          }
+        ]
+      }
+    }
+  })";
+
+  // Emulate user provided config file.
+  auto path = exec::test::TempFilePath::create();
+  auto pathStr = path->getPath();
+  createConfigFile(json, pathStr);
+
+  auto registeredFnSignaturesBefore = velox::getFunctionSignatures();
+
+  DynamicLibraryLoader testDynamicLibraryLoader(pathStr, "/");
+
+  // Read and register functions in that file. Emulating the dynamic loading
+  // process.
+  UdfBoolDynamicFnRegistration("mock1");
+  UdfVarcharDynamicFnRegistration("mock2");
+
+  auto registeredFnSignaturesAfter = velox::getFunctionSignatures();
+
+  assertFunctionExistsAfter(
+      registeredFnSignaturesBefore,
+      registeredFnSignaturesAfter,
+      "presto.default.mock1");
+  assertFunctionExistsAfter(
+      registeredFnSignaturesBefore,
+      registeredFnSignaturesAfter,
+      "presto.default.mock2");
+
+  // Config validations.
+  EXPECT_EQ(
+      testDynamicLibraryLoader.compareConfigWithRegisteredFunctionSignatures(
+          registeredFnSignaturesBefore),
+      0);
+}
+
+TEST_F(DynamicFunctionRegistererTest, newNamespace) {
+  std::string_view json = R"(
+  {
+    "dynamicUdfSignatureMap": {
+      "tmp": {
+        "mock3": [
+          {
+            "fileName": "test",
+            "nameSpace": "test.namespace",
+            "outputType": "boolean",
+            "paramTypes": [
+            ]
+          }
+        ]
+      }
+    }
+  })";
+
+  // Emulate user provided config file.
+  auto path = exec::test::TempFilePath::create();
+  auto pathStr = path->getPath();
+  createConfigFile(json, pathStr);
```

Comment:
The TempFile can be created inside createConfigFile itself, to avoid repetition. Is there any issue with that ?